lib/cdo/aws/s3.rb

```diff
@@ -69,11 +79,47 @@ def self.create_client
       connect_v2!
     end
 
+    # Populates locally hosted buckets for development environments without full AWS access.
+    def self.populate_local_bucket(bucket, key)
+      return unless CDO.aws_s3_emulated?
+      return if exists_in_bucket(bucket, key)
+
+      # Determine the Populator class that can generate files for this bucket, if it
```

Comment:
This logic (ie, find the right path and class if they exist) feels more like a utility that it would be nice to house alongside the other populator code, rather than nested in here. I'm not sure of the ruby-preferred way to do that though?

Reply:
"ruby-preferred" depends on you! what do you think it should be?